### PR TITLE
Updated to allow all devices and seek to provision, this is to suppor…

### PR DESCRIPTION
**Pull request by dashcraft**
### Primarily:
Updated to allow all devices and seek to provision, this is to support other advertising types, not just primary general advertising.

Also, there are some BLE implementations that aren't consistent in advertising generally and this results in a correct primary service advertisement using gatt, but otherwise the device would get filtered out because of it not consistently advertising its primary service.

Also, secondarily, filtering via improv only, means someone can walk around with a laptop and attempt to connect to all improv, they get an exact list of all devices that could be exploited without credentials. It's terrible security but it is making it much easier for someone to exploit, without code.

### Secondarily
I also added a build script for developers not using docker

### CODE REVIEW

File: src/provision.ts [hunk 1]
```diff
@@ -7,13 +7,15 @@ export const startProvisioning = async (button: LaunchButton) => {
   let device: BluetoothDevice | undefined;
   try {
     device = await navigator.bluetooth.requestDevice({
-      filters: [{ services: [IMPROV_BLE_SERVICE] }],
+      acceptAllDevices: true,
+      optionalServices: [IMPROV_BLE_SERVICE]
```
Comment (balloob) [inline src/provision.ts:11]: This is not a good default setting and we should not merge it.

Do you have an example of a bad BLE implementation that is bad at broadcasting it's primary service? If it's a common problem we can consider adding a config option to make "acceptAll" an option.

Comment (dashcraft) [reply src/provision.ts:11]: At the very least an "accept all" option would be great.

An example use case would be anytime I don't want improv to be my primary service, but a secondary service.
I am using nimble on a different project and there are several bugs related to primary services either getting cached, not working on chrome/safari, not broadcasting primary service consistently, or not advertising consistently.

Bluedroid doesn't seem to have an issue, it seems to be _very_ consistent in its advertisements.

Comment (dashcraft) [reply src/provision.ts:11]: I can update this to allow an options hash for defaults, if that's the direction you'd like to go.

Comment (balloob) [reply src/provision.ts:11]: Yes please make it opt-in to allow all by checking `button.hasAttribute('bluetooth-accept-all')`.

But you will also need to handle the case where a device is picked that does not have the required service.

Comment (dashcraft) [reply src/provision.ts:11]: Can do! I'll see if i can get that over in the next few days!

Btw i appreciate these standards and this service, i cannot sing the praises of hassio/esphome/nabu casa enough. Fantastic companies and fantastic developers!

**Comment by balloob** [comment]
Stale.

**Comment by dashcraft** [comment]
Yes i'll re-issue whenever i have time, I've been very busy.

**Pull request closed without merging**